Analysis/macros/templates_maker.py

```diff
@@ -217,11 +218,104 @@ def readWs(self,options,args):
+                        canv_allmlog = ROOT.TCanvas("%slog" %(title),"%slog"%title)
+                        pad2=ROOT.TPad("pad2", "pad2", 0, 0.0, 1, 0.15)
+                        pad1 = ROOT.TPad("pad1", "pad1", 0., 0.0, 1., 1.0)
+                        pad1.SetBottomMargin(0)
+                        pad1.SetLogy()
+                        pad1.Draw()
+                        pad1.cd()
+                        leg = ROOT.TLegend(0.5,0.6,0.9,0.9)
+                        leg.SetFillColor(ROOT.kWhite) 
+                    #self.workspace_.Print()
+                        isovar=self.workspace_.var("templateNdim%dDim%d" % ( fit["ndim"],idim))
+                        template_binning = array.array('d',comparison.get("template_binning",fit["template_binning"]))
+                        templatebins=ROOT.RooBinning(len(template_binning)-1,template_binning,"templatebins" )
+                        isovar.setBinning(templatebins)
+                        print isovar
+                        isoframe=isovar.frame()
+                        isoframe.SetTitle("%slog" % title)
+                        truth.plotOn(isoframe,RooFit.Rescale(1./truth.sumEntries()),RooFit.Binning(templatebins),RooFit.MarkerStyle(20),RooFit.MarkerColor(ROOT.kRed+1),RooFit.LineColor(ROOT.kRed+1),RooFit.Name(truth.GetTitle()))
+                        i=0
+                        for temp in templates:
+                            i+=2
+                            temp.plotOn(isoframe,RooFit.Rescale(1./temp.sumEntries()),RooFit.Binning(templatebins),RooFit.MarkerStyle(20),RooFit.MarkerColor(ROOT.kGreen+i),RooFit.LineColor(ROOT.kGreen+i),RooFit.Name(temp.GetTitle()))
+                        isoframe.Draw()
+                        leg.AddEntry(truth.GetTitle(),truth.GetTitle(),"l")  
+                        for temp in templates:
+                          leg.AddEntry(temp.GetTitle(),temp.GetTitle(),"l");
+                        leg.Draw()
+                        isoframe.SetAxisRange(1e-3,20,"Y")
+                        pad1.Update()
+                        isoarg=ROOT.RooArgList("isoarg")
+                        isoarg.add(isovar)
+                        truthHisto=ROOT.TH1F("%sHisto" % truth.GetTitle(),"%sHisto" % truth.GetTitle(),len(template_binning)-1,template_binning)
+                        truth.fillHistogram(truthHisto,isoarg)
+                        truthHisto.GetXaxis().SetLimits(min(template_binning),max(template_binning))
+                        truthHisto.GetYaxis().SetLimits(0.1,10.)
+                        j=0
+                        pad2.SetBottomMargin(0.3)
+                        pad2.SetTicks(0,2)
+                        pad2.SetTicky()
+                        pad2.Draw()
+                        pad2.cd()
+                        pad2.Update()
+                        ROOT.gStyle.SetOptStat(111111)
+                        ROOT.gStyle.SetOptTitle(0)
+                        for temp in templates:
+                            j+=2
```

Comment:
Hi Pasquale,

because I can distinguish the colors better that way :)
Cheers Milena


On 13 May 2015, at 17:38, musella <notifications@github.com<mailto:notifications@github.com>> wrote:


In Analysis/macros/templates_maker.py<https://github.com/musella/diphotons/pull/2#discussion_r30243859>:

> +                        isoarg.add(isovar)
> +                        truthHisto=ROOT.TH1F("%sHisto" % truth.GetTitle(),"%sHisto" % truth.GetTitle(),len(template_binning)-1,template_binning)
> +                        truth.fillHistogram(truthHisto,isoarg)
> +                        truthHisto.GetXaxis().SetLimits(min(template_binning),max(template_binning))
> +                        truthHisto.GetYaxis().SetLimits(0.1,10.)
> +                        j=0
> +                        pad2.SetBottomMargin(0.3)
> +                        pad2.SetTicks(0,2)
> +                        pad2.SetTicky()
> +                        pad2.Draw()
> +                        pad2.cd()
> +                        pad2.Update()
> +                        ROOT.gStyle.SetOptStat(111111)
> +                        ROOT.gStyle.SetOptTitle(0)
> +                        for temp in templates:
> +                            j+=2


why += 2 and not 1?

—
Reply to this email directly or view it on GitHub<https://github.com/musella/diphotons/pull/2/files#r30243859>.